python/tests/test_pyvw.py

```diff
@@ -192,4 +193,95 @@ def test_parse():
 
     ex = model.parse(["| a:1 b:0.5", "0:0.1:0.75 | a:0.5 b:1 c:2"])
     assert len(ex) == 2
+
+    # To test the prediction on multiline examples
+    finish = model.finish_example(ex)
+    assert model.predict(ex) == [0.0, 0.0]
     del model
+
+def test_namespace_id():
+    vw_ex = vw(quiet=True)
+    ex = vw_ex.example('1 |a two features |b more features here')
+    nm1 = pyvw.namespace_id(ex, 2)
+    assert nm1.id == 2
+    assert nm1.ord_ns == 128
+    assert nm1.ns == '\x80' # Represents string of ord_ns
+
+def test_example_namespace():
+    vw_ex = vw(quiet=True)
+    ex = vw_ex.example('1 |a two features |b more features here')
+    nmid = pyvw.namespace_id(ex, 1)
+    exm = pyvw.example_namespace(ex, nmid, ns_hash=vw_ex.hash_space(nmid.ns))
+    assert isinstance(exm.ex, pyvw.example)
+    assert isinstance(exm.ns, pyvw.namespace_id)
+    assert exm.ns_hash == 2514386435
+    assert exm.num_features_in() == 3
+    assert exm[2] == (11617, 1.0) # represents (feature, value)
+    assert exm.iter_features()
+    assert exm.pop_feature()
+    exm.push_features(nmid, ['c', 'd'])
+    assert exm.num_features_in() == 4
+
+def test_simple_label():
+    sl = pyvw.simple_label(2.0, weight=0.5)
+    assert sl.label == 2.0
+    assert sl.weight == 0.5
+    assert sl.prediction == 0.0
+    assert sl.initial == 0.0
+    assert str(sl) == '2.0:0.5'
+
+def test_simple_label_example():
+    vw_ex = vw(quiet=True)
+    ex = vw_ex.example('1 |a two features |b more features here')
+    sl2 = pyvw.simple_label(ex)
+    assert sl2.label == 1.0
+    assert sl2.weight == 1.0
+    assert sl2.prediction == 0.0
+    assert sl2.initial == 0.0
+    assert str(sl2) == '1.0'
+
+def test_multiclass_label():
+    ml = pyvw.multiclass_label(2, weight=0.2)
+    assert ml.label == 2
+    assert ml.weight == 0.2
+    assert ml.prediction == 1
+    assert str(ml) == '2:0.2'
+
+def test_multiclass_label_example():
+    vw_ex = vw(quiet=True)
+    ex = vw_ex.example('1 |a two features |b more features here')
+    ml2 = pyvw.multiclass_label(ex)
+    assert ml2.label == 1065353216
+    assert ml2.weight == 1.0
+    assert ml2.prediction == 0
+    assert str(ml2) == '1065353216'
+
+def test_example():
```

Comment:
There is a lot of functionality being tested here. Ideally it would be better in different tests please

Reply:
Sure, I will part them up. Thanks!